src/relation_engine_server/api.py

```diff
@@ -33,9 +34,13 @@ def run_query():
         del json_body['query']
         resp_body = arango_client.run_query(query_text=query_text, bind_vars=json_body)
         return flask.jsonify(resp_body)
-    # auth.require_auth_token(roles=[])
     if 'view' in flask.request.args:
         # Run a query from a view name
+        json_body['ws_ids'] = []
+        auth_token = auth.get_auth_header()
+        if auth_token:
+            # Handle workspace authentication
+            json_body['ws_ids'] = auth.get_workspace_ids(auth_token)
```

Comment:
I think we want this to run with adhoc queries as well to better represent the behavior of a defined view

Reply:
Good call on the adhoc queries, didn't consider those, will add that.

Reply:
Having read-everything admin access adds another whole layer that I'm still working on, so this PR is not really complete. Will let you know when the pr is ready again.